### PR TITLE
Decouple the configmap from the chart 

### PR DESCRIPTION
**Pull request by mandopaloooza**
Operator synced with ArgoCD is always reporting "Out of Sync" because the configmap (which part of the chart) is constantly being updated by the operator to keep a record of entity metadata. I removed the configmap from the chart and programmatically create the configmap so it doesn't become a target for syncing by helm/argocd.

### CODE REVIEW

**Review by thomascellerier — APPROVED** [review]

File: appgate/client.py [hunk 1]
```diff
@@ -238,12 +243,42 @@ async def init(self) -> None:
             self.namespace,
             self.name,
         )
+
+        def configmap_exists() -> bool:
+            configmaps = self._v1.list_namespaced_config_map(namespace=self.namespace)  # type: ignore
```
[the remark this answers is not in the document]
Comment (mandopaloooza) [reply appgate/client.py:248]: Actually, I tried https://pypi.org/project/kubernetes-typed/ but I would need to drop the mypy version to 0.910 

```
There are incompatible versions in the resolved dependencies:
  mypy==0.991 (from -r requirements-build.in (line 1))
  mypy<=0.910,>=0.800 (from kubernetes-typed==18.20.0->-r requirements-build.in (line 9))
```

[the remark this answers is not in the document]
Comment (mandopaloooza) [reply appgate/client.py:248]: Match statement got supported in 0.940...

[the remark this answers is not in the document]
Comment (thomascellerier) [reply appgate/client.py:248]: Ah well nevermind for now :).

File: README.md [hunk 2]
```diff
@@ -90,7 +90,6 @@ Browse the available charts versions in the [SDP Operator GitHub Container Regis
 | `sdp.operator.image.pullSecrets` | The secret to access the repository.                                                              | `[]`                           |
 | `sdp.operator.fernetKey`         | The fernet key to use when decrypting secrets in entities.                                        | `""`                           |
 | `sdp.operator.caCert`            | The controller's CA Certificate in PEM format. It may be a base64-encoded string or string as-is. | `""`                           |
-| `sdp.operator.configMapMt`       | The config map to store metadata for entities.                                                    | `""`                           |
```
Comment (lepiolet) [inline README.md:93]: :+1:

File: appgate/client.py (hunk 1 above)
Comment (lepiolet) [inline appgate/client.py:248]: can't we query directly for the configmap with name in ns?

File: appgate/client.py [hunk 3]
```diff
@@ -238,12 +243,42 @@ async def init(self) -> None:
...
+            for configmap in configmaps.items:
+                if configmap.metadata.name == self.name:
+                    return True
+            return False
+
+        def initialize_configmap() -> V1ConfigMap:
+            if configmap_exists():
```
Comment (lepiolet) [inline appgate/client.py:255]: I would do some `get_configmap` returning `Optional`, then you just check if it's None or not

File: k8s/operator/templates/rbac.yaml [hunk 4]
```diff
@@ -88,8 +88,8 @@ rules:
       - serviceusers
     verbs: ["get", "watch", "list"]
   - apiGroups: [""]
-    resources: ["configmaps", "secrets"]
```
Comment (lepiolet) [inline k8s/operator/templates/rbac.yaml:91]: :+1:

**Pull request merged**